src/App.tsx

```diff
@@ -1,34 +1,149 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { ChangeEvent } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { FilterStatus } from './types/FilterStatus';
+import { TodoFilter } from './components/TodoFilter';
+import * as api from './api';
+
+type State = {
+  todos: Todo[],
+  user: User | null,
+  query: string,
+  filter: FilterStatus,
+  selectedTodoId: number | null,
+  selectedUserId: number | null,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    todos: [],
+    user: null,
+    query: '',
+    filter: FilterStatus.all,
+    selectedTodoId: null,
+    selectedUserId: null,
+  };
+
+  async componentDidMount() {
+    const todos = await api.getTodos();
+
+    this.setState({ todos });
+  }
+
+  loadUser = async () => {
+    if (this.state.selectedUserId !== null) {
+      // eslint-disable-next-line
+      const user = await api.getUser(this.state.selectedUserId);
+
+      this.setState({ user });
+    }
+  };
+
+  render() {
+    const {
+      todos,
+      user,
+      query,
+      filter,
+      selectedTodoId,
+      selectedUserId,
+    } = this.state;
+    const lowQuery = query.toLowerCase().trim();
+    let visibleTodos = [...todos];
+
+    const handleChangeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+      return this.setState({ query: event.target.value });
+    };
 
-export const App: React.FC = () => {
-  return (
-    <>
+    const handleQueryReset = () => {
+      return this.setState({ query: '' });
+    };
+
+    const handleFilterChange = (event: ChangeEvent<HTMLSelectElement>) => {
+      return this.setState({ filter: event.target.value as FilterStatus });
+    };
+
+    const changeTodo = (todoId: number | null) => {
+      this.setState({ selectedTodoId: todoId });
+    };
+
+    const changeUser = (userId: number | null) => {
+      this.setState({ selectedUserId: userId });
+    };
+
+    const resetUser = () => {
+      this.setState({ user: null });
+    };
+
+    if (query[0] === ' ') {
+      this.setState({ query: '' });
+    }
+
+    if (filter === FilterStatus.active) {
+      visibleTodos = todos.filter(
+        (todo: Todo) => todo.completed === false,
+      );
+    }
+
+    if (filter === FilterStatus.completed) {
+      visibleTodos = todos.filter(
+        (todo: Todo) => todo.completed === true,
+      );
+    }
+
+    visibleTodos = visibleTodos.filter(
+      (todo: Todo) => todo.title.toLocaleLowerCase().includes(lowQuery),
+    );
```

Comment:
you shouldn't put this code here, in the render function
![image](https://user-images.githubusercontent.com/59209337/234494647-c9cad653-a864-4648-9cae-ea646094cdd4.png)